### PR TITLE
fix(restore): adding check for CVR creation before making the restore REST API call

### PR DESCRIPTION
**Pull request by mynktl**
Changes:
- Added check for CVR creation before making the restore REST API call.
  plugin will wait until all required CVR for given volumes are created and in healthy/degraded/error state.
- Refactored the code for readability.
- Added error handling for PV having released state. `GetVolumeID` will return an error if given PV is in the released state.

Signed-off-by: mayank <mayank.patel@mayadata.io>

### CODE REVIEW

File: pkg/cstor/cvr_operation.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import (
+	"github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// waitForAllCVR will ensure that all CVR related to
+// given volumes are created
+func (p *Plugin) waitForAllCVR(vol *Volume) error {
```
Comment (vishnuitta) [inline pkg/cstor/cvr_operation.go]: should we have a timelimit or iterations limit here?
Also, seeing tight loop which can be fixed

Comment (vishnuitta) [reply pkg/cstor/cvr_operation.go]: Add a log as well to see what is happening

File: pkg/cstor/pvc_operation.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		if pvc.Status.Phase == v1.ClaimBound {
+			p.Log.Infof("PVC(%v) created..", pvc.Name)
+			vol := &Volume{
+				volname:    pvc.Spec.VolumeName,
+				namespace:  pvc.Namespace,
+				backupName: snapName,
+				casType:    *pvc.Spec.StorageClassName,
+			}
+
+			if err = p.waitForAllCVR(vol); err != nil {
```
Comment (vishnuitta) [inline pkg/cstor/pvc_operation.go]: we can move this out of for loop as part of code reorg

File: pkg/cstor/pvc_operation.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	pvc.Annotations["openebs.io/created-through"] = "restore"
+	rpvc, err := p.K8sClient.
+		CoreV1().
+		PersistentVolumeClaims(pvc.Namespace).
+		Create(&pvc)
+	if err != nil {
+		return nil, errors.Errorf("Failed to create PVC : %s", err.Error())
+	}
+
+	for {
```
Comment (vishnuitta) [inline pkg/cstor/pvc_operation.go]: here as well, there is tight loop

File: pkg/cstor/pvc_operation.go (hunk 2 above)
Comment (mynktl) [reply pkg/cstor/pvc_operation.go]: yes

File: pkg/cstor/pvc_operation.go (hunk 3 above)
Comment (mynktl) [reply pkg/cstor/pvc_operation.go]: yes. I will remove it.

File: pkg/cstor/cvr_operation.go (hunk 1 above)
Comment (mynktl) [reply pkg/cstor/cvr_operation.go]: Done

**Pull request merged**